app/controllers/application_controller.rb

```diff
@@ -49,6 +49,11 @@ class ApplicationController < Sinatra::Base
     { payload: }.to_json
   end
 
+  patch '/cleanup' do
+    Bin.cleanup
+    status 204
+  end
```

Comment:
What happens if someone calls that router very fast very often when the database is very big?

I would suggest to create some kind of locking mechanism

Reply:
Expired bins are now deleted

* in batches (of 1000) - see https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-in_batches
* with locking - see https://api.rubyonrails.org/classes/ActiveRecord/Locking/Pessimistic.html
* async - with sucker_punch gem